x-pack/plugins/ml/server/models/results_service/anomaly_charts.ts

```diff
@@ -1124,9 +1124,17 @@ export function anomalyChartsDataProvider(mlClient: MlClient, client: IScopedClu
         each(scheduledEvents, (events, time) => {
           const chartPoint = findChartPointForTime(chartDataForPointSearch, Number(time));
           if (chartPoint !== undefined) {
-            // Note if the scheduled event coincides with an absence of the underlying metric data,
-            // we don't worry about plotting the event.
             chartPoint.scheduledEvents = events;
+          } else {
+            // If there's no underlying metric data point for the scheduled event,
+            // create a new chart point with a value of 0.
+            const eventChartPoint: ChartPoint = {
+              date: Number(time),
+              value: 0,
+              scheduledEvents: events,
+              isFakeDataPoint: true,
+            };
+            chartData.push(eventChartPoint);
```

Comment:
Looks like this might be unnecessarily adding extra scheduled event markers into the single metric type of charts:

<img width="861" alt="Screenshot 2024-09-10 at 12 27 46" src="https://github.com/user-attachments/assets/9ba4e064-6bb8-42c8-af95-774d21cce723">

Reply:
Retested against https://github.com/elastic/kibana/pull/192377/commits/01c4d41136248f349eafc1b9bb29265c7ce77d1a and this looks fixed.